kpatch-build/create-diff-object.c

```diff
@@ -1923,6 +1923,28 @@ static void kpatch_regenerate_special_section(struct kpatch_elf *kelf,
 	sec->base->data->d_size = dest_offset;
 }
 
+static void kpatch_check_relocations(struct kpatch_elf *kelf)
+{
+	struct rela *rela, *safe;
+	struct section *sec;
+	Elf_Data *sdata;
+
+	list_for_each_entry(sec, &kelf->sections, list) {
+		if (!is_rela_section(sec))
+			continue;
+		list_for_each_entry_safe(rela, safe, &sec->relas, list) {
+			if (rela->sym->sec) {
+				sdata = rela->sym->sec->data;
+				if (rela->addend < sdata->d_off ||
+						rela->addend > (sdata->d_off + sdata->d_size)) {
```

Comment:
Realistically, I believe there's always only *one* data descriptor per section and `d_off` will always be zero, so it should be fine to do `if (rela->addend >= sdata->d_size)`.

Reply:
Should this be >= or >? I've been testing with the patch from #615 on top of current Linus' tree and am getting a relocation that would fail >= check. The resulting module does load alright though:

```
  [12] .text.compat_SyS_getdents PROGBITS        0000000000000000 0002d0 00012f 00  AX  0   0 16
                                                                                   
Relocation section '.rela.debug_ranges' at offset 0x39850 contains 34 entries:  
    Offset             Info             Type               Symbol's Value  Symbol's Name + Addend
...                                                                                
00000000000004d8  0000000a00000001 R_X86_64_64            0000000000000000 .text.compat_SyS_getdents + 12f 
...                                                                                

```

Reply:
Ah, that's one of the DWARF debug sections.  I think '>' is probably fine then.